### PR TITLE
Added Deliverables and Modified Section 4.3.2 #374 #375

### PR DESCRIPTION
**Pull request by densellp**
- Removed exact instructions for creating a VM to encourage learning
- Added deliverables for users to discuss

### CODE REVIEW

**Comment by densellp** [comment]
- Are there other ways to authenticate with a virtual machine instance other than using already generated ssh keys?
* Yes, there is an option to generate-keys when creating the VM, which will already create and use the ssh key when you attempt to login, if you already have your own keys, you can specify them in the creation of the VM
- What is cloud-init, and why is it an industry standard?
* Cloud init is a multi platform, multi distribution cloud instance initialisation. A majority if not all the cloud providers support cloud init in some way. Cloud Init will setup SSH keys, packages, networking, storage, etc. on boot. 
- What are the specific Azure CLI query commands that can be used to assign the public IP into a variable?
* You can run a command like this ```az vm show --resource-group MyResourceGroup --name MyVM --query 'networkProfile.networkInterfaces[0].id' --output tsv``` which allows you to grab the IP address of a vm once you query it by name. To set to an environment variable, just save it.
- How can the process be automated or streamlined for efficient deployment and management?
* Terraform is a great tool that would allow developers to take all of this configuration and be able to create and provision Virtual machines, and other resources, the cloud init file can still be passed in as a custom data script once the VM is created. Terraform can also tear down all the resources as if they were never there with a simple command.

**Comment by densellp** [comment]
Section should only take about 240 minutes, since the cloud init script is already given to you and the vm creation isn't all that bad, just reading some docs and making sure your configuration is correct.

File: docs/4-cloud-computing/4.3.2-virtual-machines.md [hunk 1]
```diff
@@ -70,39 +69,39 @@ runcmd:
...
-    --image UbuntuLTS \
-    --admin-username azureuser \
-    --ssh-key-values <path to SSH public key from step 4> \
-    --custom-data /Users/YOUR_NAME/cloud-init.txt
+az vm -h
 ```
 
 ?> It might take several minutes for cloud-init to install packages and perform updates. This is especially true for small instance sizes!
 
+?> You may need to pass in the config file as base 64 encoded instead of txt
```
Comment (jburns24) [inline docs/4-cloud-computing/4.3.2-virtual-machines.md:86]:
```suggestion
?> You may need to pass in the config file as base64 encoded
```

File: docs/4-cloud-computing/4.3.2-virtual-machines.md [hunk 2]
```diff
@@ -70,39 +69,39 @@ runcmd:
...
+
 7. Use Azure CLI to get the public IP for your new VM. For bonus points, use Azure CLI query commands to assign the public IP into a variable we can use in subsequent steps. You can [look here](https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest) for VM specific CLI command references.
 
-8. Make sure you can SSH into the VM using the private key you generated in step 4.
+8. Make sure you can SSH into the VM using the private key from step 4.
 
 9. When the VM was deployed a default NSG was attached to the VM's NIC. Use Azure CLI to open up web traffic to the VM.
 
-10. Test you can reach the website using cURL.
+10. Test you can reach the website using CURL.
```
Comment (jburns24) [inline docs/4-cloud-computing/4.3.2-virtual-machines.md:94]:
```suggestion
10. Test you can reach the website using cURL.
```
technically written as [cURL](https://en.wikipedia.org/wiki/CURL)

**Review by jburns24 — APPROVED** [review]

**Pull request merged**